api/envoy/api/v2/core/protocol.proto

```diff
@@ -216,6 +229,29 @@ message Http2ProtocolOptions {
   //
   // See `RFC7540, sec. 8.1 <https://tools.ietf.org/html/rfc7540#section-8.1>`_ for details.
   bool stream_error_on_invalid_http_messaging = 12;
+
+  // Specifies SETTINGS frame parameters to be sent to the peer, with one exception:
+  // SETTINGS_ENABLE_PUSH (0x2) is not configurable as HTTP/2 server push is not supported by Envoy.
+  //
+  // Note that custom parameters specified through this field may not collide with the following
+  // named parameters:
+  //
+  // .. code-block:: text
+  //
+  //   ID    Field Name
+  //   ----------------
+  //   0x1   hpack_table_size
+  //   0x3   max_concurrent_streams
+  //   0x4   initial_stream_window_size
+  //
+  // Collisions will trigger config validation failure on load/update.
+  // The only exception is '0x8 allow_connect', for which the named parameter field will be
+  // overridden by the value set in this field.
```

Comment:
should we deprecate that field to avoid the confusion in the next api rev?
If so let's update the docs to make this the default way to configure etc.

Also if we can consider it a failure to configure both I would - best to avoid confusion.

Reply:
What is the deprecation process to follow? I found https://github.com/envoyproxy/envoy/blob/master//CONTRIBUTING.md#breaking-change-policy, and given the level of effort I would prefer to introduce the deprecation in a follow up PR if possible to avoid holding this up.

Reply:
Yes +1 on let's do this in a follow up. I would like to land this PR and it's complex enough as it is. I agree with @alyssawilk that we should do this. Can you open an issue to track?

Reply:
SG. Filed https://github.com/envoyproxy/envoy/issues/10412.

Reply:
cool, I'm fine with a follow-up for deprecation but I'd be inclined to mention the plan in the API comments as well.
I'd also still suggest we disallow it being set both ways for the 2 years until we can remove the deprecated field, which I think should land with this PR unless it's super complicated.  Thoughts there?

Reply:
Just to clarify, is your proposal that `allow_connect` should only be allowed to be configured via a custom parameter?